client/src/components/Search.js

```diff
@@ -101,36 +81,41 @@ function Search(props) {
         const resortFilter = parameters.get('resort');
         const languageFilter = parameters.get('language');
 
-        //if the filters have been reset - reset also the filters in the app
-        props.getResortSearched("")
-        props.getLanguageSearched("")
+        //if the dropbox content has been cleared - reset also the filters in the app - Parent component
+        props.setResortSearched("")
+        props.setLanguageSearched("")
+        props.setSportSearched("")
+        props.setLevelSearched("")
    
         if (sportFilter) {
             setSport(sportFilter)
-            props.getSportSearched(sportFilter)
+            if (sports.includes(sportFilter)) {
+                props.setSportSearched(sportFilter)
+            } 
         }
         if (levelFilter) {
             setLevel(levelFilter)
-            props.getLevelSearched(levelFilter)
+            if (levels.includes(levelFilter)) {
+                props.setLevelSearched(levelFilter)
+            } 
+           
         }
         if (resortFilter) {
             setResort(resortFilter)
             
             if(props.skierListData.length > 0) {
                 if (resortSuggestions.includes(resortFilter)) {
-                    props.getResortSearched(resortFilter)
+                    props.setResortSearched(resortFilter)
                 } 
             }
         }
         if (languageFilter) {
-            //not working proprely - to check
             setLanguage(languageFilter)
             if (languageSuggestions.includes(languageFilter)) {
-                props.getLanguageSearched(languageFilter)
+                props.setLanguageSearched(languageFilter)
             }
```

Comment:
@giacomorebonato ,

I'm looking also at these lines and it's all being very repetitive. Do you think there should be a custom hook for these lines too? I tried to put them in a function but I couldn't yet figure it out. 

Thank you!
Stefi